### ISSUE TITLE
to_pandas handling Date types

### ISSUE THREAD

**Issue by ChristopherRussell** [pola-rs/polars#6922]
### Problem description

`to_pandas` converts date types to object. I don't think this is a bug because it is consistent with how pandas treats a dataframe column containing `datetime.date`. See example below. However the result is seems undesirable as the object column is far less useful than a `datetime` column would be.

Would it be reasonable for `to_pandas` to (possibly optionally) cast `pl.Date` to `datetime`?

```
import pandas as pd
import polars as pl
import datetime
df_polars = pl.DataFrame({
    'date': [datetime.date(2023,2,15)]
})
df_pandas = pd.DataFrame({
    'date': [datetime.date(2023,2,15)]
})
print(f"Polars dtype: {df_polars .schema['date']}")
print(f"Polars.to_pandas dtype: {df_polars .to_pandas().date.dtype}")
print(f"Pandas dtype: {df_pandas.date.dtype}")


Polars dtype: Date
Polars.to_pandas dtype: object
Pandas dtype: object
```

**Comment by ritchie46** [comment]
This has been fixed in master. I will release tonight.

**Comment by cmdlineluser** [comment]
https://github.com/pola-rs/polars/issues/6900 is the previous issue for reference.

**Comment by alexander-beedie** [comment]
@ChristopherRussell; this was an accidental regression in the previous release; the fix will be released later today (keep an eye out for `0.16.6` :)

In the meantime you can guarantee the desired/previous behaviour like so:
```python
df_polars.to_pandas( date_as_object=False )
```
